contracts/cairo/src/tests/test_escrow.cairo

```diff
@@ -189,4 +190,179 @@ mod Escrow {
+        stop_prank(CheatTarget::One(escrow.contract_address));
+
+        start_prank(CheatTarget::One(escrow.contract_address), USER());
+        let order = Order { recipient_address: 12345.try_into().unwrap(), amount: 500, fee: 0 };
+        let order_id = escrow.set_order(order);
+        stop_prank(CheatTarget::One(escrow.contract_address));
+    }
+
+    #[test]
+    fn test_interact_when_unpaused() {
+        let (escrow, _) = setup();
+
+        start_prank(CheatTarget::One(escrow.contract_address), USER());
+        let order = Order { recipient_address: 12345.try_into().unwrap(), amount: 500, fee: 0 };
+        let order_id = escrow.set_order(order);
+        stop_prank(CheatTarget::One(escrow.contract_address));
+
+        start_prank(CheatTarget::One(escrow.contract_address), OWNER());
+        escrow.pause();
+        escrow.unpause();
+        stop_prank(CheatTarget::One(escrow.contract_address));
+
+        start_prank(CheatTarget::One(escrow.contract_address), USER());
+        let order = Order { recipient_address: 12345.try_into().unwrap(), amount: 500, fee: 0 };
+        let order_id = escrow.set_order(order);
+        stop_prank(CheatTarget::One(escrow.contract_address));
+    }
+
+    #[test]
+    fn test_upgrade_when_paused() {
+        let (escrow, _) = setup();
+        let upgradeable = IUpgradeableDispatcher { contract_address: escrow.contract_address };
+
+        start_prank(CheatTarget::One(escrow.contract_address), OWNER());
+        escrow.pause();
+        upgradeable.upgrade(declare('Escrow_mockPausable').class_hash);
+
+        let escrow_2 = IEscrow_mockPausableDispatcher { contract_address: escrow.contract_address };
+        assert(escrow_2.pause_state() == true, 'Contract should be paused');
+
+        stop_prank(CheatTarget::One(escrow.contract_address));
+    }
+
+    #[test]
+    fn test_fail_l1_while_paused() {
+        let (escrow, _) = setup();
+        start_prank(CheatTarget::One(escrow.contract_address), OWNER());
+        escrow.pause();
+        stop_prank(CheatTarget::One(escrow.contract_address));
+
+        let data: Array<felt252> = array![1, MM_ETHEREUM().into(), 3, 4];
+        let mut payload_buffer: Array<felt252> = ArrayTrait::new();
+        data.serialize(ref payload_buffer);
+        let mut l1_handler = L1HandlerTrait::new(
+            contract_address: escrow.contract_address,
+            function_name: 'withdraw',
+        );
+        l1_handler.from_address = ETH_TRANSFER_CONTRACT().into();
+        l1_handler.payload = payload_buffer.span();
+
+        // same as "Should Panic" but for the L1 handler function
+        match l1_handler.execute() {
+            Result::Ok(_) => panic_with_felt252('shouldve panicked'),
+            Result::Err(RevertedTransaction) => {
+                assert(*RevertedTransaction.panic_data.at(0) == 'Pausable: paused', *RevertedTransaction.panic_data.at(0));
+            }
+        }
+    }
+
+    #[test]
+    fn test_fail_l1_from_user() {
```

Comment:
exactly. 
Once having cracked the calling of the L1 handler, i saw the opportunity to add this test to assure only YABTransfer can call Escrow's Withdraw function
Beware this is different from the ACL implemented in YABTransfer

Reply:
ok then let's change the name because it's not clear enough. It should be something that indicates how this fails. In this case we are expecting that only YABTransfer can call this l1 handler.